.github/workflows/publish.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r dev_requirements.txt
+        pip install -r requirements.txt
+    - name: Build
+      run: |
+        python setup.py bdist_wheel --python-tag py3
+    - name: Upload
+      env:
+        TWINE_USERNAME: __token__
+        TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+      run: |
+        twine upload dist/*
```

Comment:
no guarantee that things we publish are tested

Reply:
I.e. the test workflow isn't guaranteed to have returned green for this to run
But that's easy to fix

Reply:
I added it first but it was ugly to write it twice. Only looked into depending on another workflow very briefly but there's probably an okay solution

Reply:
I can count how many times I have published bad packages on my hand (even once that customers have got in there lap) which is kinda unnecessary
So I have it as a memo in the back of my head for publishing
1. make absolutely sure that its the correct version of the code (i.e. no caching etc)
1. make absolutely sure that its the correct dependencies installed for it
1. always be 100% sure that the lib/bin is tested
1. if possible ensure that the lib/bin is installable/runnable

Reply:
I would say `dependable > ugly` when it comes to this

Reply:
also if you test it here you test it without the cache so you are more sure that the tests just works because of some cache somewhere

Reply:
We are hashing the requirements files so I think we should be ok on the caching

Reply:
if all dependencies are fixed `==`, none of the dependencies has been yanked or any other external dependencies has changed like patch version of python in the github action container etcetc

Reply:
its okay for testing, but I would say we want to test it from scratch when releasing